### DISCUSSION TITLE
Loading ABI for ETH ERC-20

### DISCUSSION THREAD

**Discussion by stanleyseow** [reach-sh/reach-lang#1541] [Q&A] [answered]
I'm trying to attach to an ERC-20 contract I created in Remix.

I have downloaded the ABI, how can I load this ABI into Reach so it can make remote contract calls via APIs ?

I searched all the related abi topics but could not find anything...

Thanks

**Comment by jeapostrophe** [comment, marked as answer]
You can use any remote object by passing an interface to the `remote` function, along with the contract address --- https://docs.reach.sh/rsh/consensus/#remote-objects

Reach can also work with ERC-20s by using the `Token` type.

**Comment by stanleyseow** [comment]
Thanks...

Let me see if I understand this correctly ..
From the ABI below for _transfer_ function, I will create a **remote object** like this :

Input : _Address, UInt256_
Output : _Bool_

```
const token =
  remote( tokenCtcInfo, {
    transferTo: [Fun] [Address, UInt, Address], Bool),
  });
```



### ERC20 ABI 

```
{    
    "name": "transfer",
    "type": "function",
    "inputs": [
        {
            "internalType": "address",
            "name": "to",
            "type": "address"
        },
        {
            "internalType": "uint256",
            "name": "amount",
            "type": "uint256"
        }
    ],
    "outputs": [
        {
            "internalType": "bool",
            "name": "",
            "type": "bool"
        }
    ],
    "stateMutability": "nonpayable",

},
```

**Reply by jeapostrophe** [reply]
There is no `transferTo` function in the ERC-20 standard. There is just `transfer`. You'd write:
```
const token =
  remote( tokenCtcInfo, {
    transfer: Fun([Address, UInt], Bool),
  });
```

**Comment by stanleyseow** [comment]
Add-on question ...

_REMOTE_FUN.withBill() --- Returns a remote function that provides some number of network tokens and, possibly, non-network tokens to the caller when it returns. The exact amount is returned from the invocation by wrapping the original result in a tuple._

How to I specify the **non-network tokens** to be billed vs **network tokens** ??

**Reply by jeapostrophe** [reply]
If you are calling ERC-20s function directly, you aren't using Reach's non-network token abstraction, so you shouldn't. But if you did want to, you write a pay amount --- https://docs.reach.sh/rsh/step/#payAmt --- which is `[amt, token]`.
